src/map/script.c

```diff
@@ -7475,16 +7476,102 @@ BUILDIN(makeitem)
 	} else
 		m=map->mapname2mapid(mapname);
 
-	if( m == -1 ) {
+	if (m == -1) {
 		ShowError("makeitem: creating map on unexistent map '%s'!\n", mapname);
+		script_pushint(st, 0);
 		return false;
 	}
 
-	memset(&item_tmp,0,sizeof(item_tmp));
+	memset(&item_tmp, 0, sizeof(item_tmp));
 	item_tmp.nameid = nameid;
-	item_tmp.identify=1;
+	item_tmp.identify = 1;
+
+	if (!itemdb->isstackable(nameid))
+		amount = 1;
+
+	script_pushint(st, ( map->addflooritem(NULL, &item_tmp, amount, m, x, y, 0, 0, 0, 0) == 1 ) ? 1 : 0);
+
+	return true;
+}
+
+/*==========================================
+* makeitem2 <item id or name>,<amount>,"<map name>",<X>,<Y>,<identify>,<refine>,<attribute>,<card1>,<card2>,<card3>,<card4>;
+*------------------------------------------*/
+BUILDIN(makeitem2) 
+{
+	int nameid, amount;
+	int x, y, m, identify, refine, attribute, c1, c2, c3, c4;
+	const char *mapname;
+	struct item item_tmp;
+	struct item_data *item_data;
+
+	if (script_isstringtype(st, 2)) {
+		const char *name = script_getstr(st, 2);
+		struct item_data *item_data = itemdb->search_name(name);
+		if (item_data)
+			nameid = item_data->nameid;
+		else
+			nameid = UNKNOWN_ITEM_ID;
+	} else {
+		nameid = script_getnum(st, 2);
+		if (nameid <= 0 || !itemdb->exists(nameid)) {
+			ShowError("makeitem2: Nonexistant item %d requested.\n", nameid);
+			script_pushint(st, 0);
+			return false; //No item created.
+		}
+	}
+
+	amount = script_getnum(st, 3);
+	mapname = script_getstr(st, 4);
+	x = script_getnum(st, 5);
+	y = script_getnum(st, 6);
+	identify = script_getnum(st, 7);
+	refine = script_getnum(st, 8);
+	attribute = script_getnum(st, 9);
+	c1 = (short) script_getnum(st, 10);
+	c2 = (short) script_getnum(st, 11);
+	c3 = (short) script_getnum(st, 12);
+	c4 = (short) script_getnum(st, 13);
+
+	if (strcmp(mapname, "this") == 0) {
```

Comment:
I just know that `*makeitem` script command can use "this" flag

from my past scripting experience, `*makeitem` used in event script usually runs without a RID attached
so "this" flag just plain stupid
also, the `*monster` support "this" flag is because it also support the -1 coordinate

`monster "this", -1, -1, "--ja--", PORING, 1, "";`
this will summon a poring NEARBY the attached player, not on the player standing on

your implementation, which copy from `*makeitem`, doesn't support this -1 coordinate input

its either you remove the "this" flag (which you also didn't mention in the document)
or make it support for the -1 input to spawn items AROUND the player without using getmapxy